### PR TITLE
[yaml] remove PubSubLite from API docs

### PR DESCRIPTION
**Pull request by Polber**
PubSubLite is deprecated as per https://cloud.google.com/pubsub/lite/docs, scheduled to be turned down March 18, 2026, and is no longer available for new customers as of September 24, 2024.

In order to avoid confusion for new Beam YAML users, this transform will be hidden from the API docs, but still usable within the YAML API

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by Polber** [comment]
R: @robertwb

**Comment by Polber** [comment]
R: @damccorm

File: sdks/python/apache_beam/yaml/generate_yaml_docs.py [hunk 1]
```diff
@@ -189,11 +189,11 @@ def io_grouping_key(transform_name):
...
-SKIP = [
-    'Combine',
-    'Filter',
-    'MapToFields',
-]
+# Deprecated providers
+SKIP = {
+    'ReadFromPubSubLite',
+    'WriteToPubSubLite',
+}
```
[the remark this answers is not in the document]
Comment (robertwb) [reply sdks/python/apache_beam/yaml/generate_yaml_docs.py:196]: That makes a lot of sense actually.

[the remark this answers is not in the document]
Comment (Polber) [reply sdks/python/apache_beam/yaml/generate_yaml_docs.py]: updated

**Comment by Polber** [comment]
Test failures seem unrelated

**Review by damccorm — APPROVED** [review]

**Pull request merged**